### PR TITLE
Biz 155 templates

### PR DESCRIPTION
**Pull request by RomanFedytskyi**

### CODE REVIEW

File: dev/components/templatesList/component.js [hunk 1]
```diff
@@ -0,0 +1,18 @@
+'use strict';
+(function (angular) {
+  angular.module('app').component('templateList', {
+    templateUrl: 'components/templatesList/tpl.html',
+    controller: ctrl,
+    controllerAs: 'vm',
+    bindings: {
+      use: '&',
```
Comment (dmitrij-borchuk) [inline dev/components/templatesList/component.js]: what is use?

File: dev/components/templatesList/component.js [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+      delete: '&',
```
Comment (dmitrij-borchuk) [inline dev/components/templatesList/component.js]: rename to 'onDelete'

File: dev/components/templatesList/style.scss [hunk 3]
```diff
@@ -0,0 +1,61 @@
+//CONTENT
+.content {
```
Comment (dmitrij-borchuk) [inline dev/components/templatesList/style.scss]: Class name is too common

File: dev/components/videoList/tpl.pug [hunk 4]
```diff
@@ -21,4 +21,6 @@
     .video-container(ng-repeat="item in vm.list")
       video-tile(
         video="item"
-        on-send-click="vm.onSendClick({$data: item})")
\ No newline at end of file
+        on-send-click="vm.onSendClick({$data: item})"
+        show-send-handler="vm.showSendHandler"
```
Comment (dmitrij-borchuk) [inline dev/components/videoList/tpl.pug]: I am not sure that it is working

File: dev/components/videoTile/tpl.pug [hunk 5]
```diff
@@ -1,9 +1,14 @@
...
+    src='data:image/png;base64,{{vm.video.attributes.thumbnail}}',
+    alt=''
+    ng-click="vm.onThumbnailClick()")
   p.text.ellipsis(title='{{vm.video.attributes.name}}') {{vm.video.attributes.name}}
   .buttons-container
     a.menu
       img(src='/files/images/dots.png', alt='')
-    .inlineBlock(ng-click="vm.onSendClick()")
+    .inlineBlock(
+      ng-if="vm.showSendHandler"
```
Comment (dmitrij-borchuk) [inline dev/components/videoTile/tpl.pug]: Is the the 'showSendHandler' function? If yes why you are checking if it exists? If not why it has 'handler' in the name?

File: dev/config.js [hunk 6]
```diff
@@ -63,6 +63,10 @@
       controller: 'profileCtrl',
       controllerAs: 'vm',
       templateUrl: 'profile/tpl.html'
+    }).when('/template/:id/:templateId', {
```
Comment (dmitrij-borchuk) [inline dev/config.js]: What is 'id'?

File: dev/conversations/tpl.pug [hunk 7]
```diff
@@ -23,8 +23,8 @@
     .videoHeader
       .photo
       .text
-        p.userName {{vm.user.name}}
-        p.userMessage {{vm.user.message}}
+        p.user-name {{vm.user.name}}
+        p.user-role {{vm.user.message}}
```
Comment (dmitrij-borchuk) [inline dev/conversations/tpl.pug]: Class name is .user-role but content is message. Something wrong

File: dev/library/ctrl.js [hunk 8]
```diff
@@ -3,27 +3,34 @@
...
     vm.showUploadPopup = false;
     vm.showSendPopup = false;
     vm.recordedData = null;
-
-    var getVideos = function () {
-      libraryService.getVideos().then(function (res) {
-        vm.list = res.data.data;
-      });
-    };
+    vm.showSendHandler = true;
```
Comment (dmitrij-borchuk) [inline dev/library/ctrl.js]: 'Handler' should be in names of the callbacks

File: dev/library/ctrl.js [hunk 9]
```diff
@@ -68,18 +75,42 @@
 
     // Send
     vm.sendClickHandler = function (video) {
-      vm.videoForSending = video;
-      vm.showSendPopup = true;
+      $location.path('template/' + video.id + '/null');
```
Comment (dmitrij-borchuk) [inline dev/library/ctrl.js]: What for 'null'?

File: dev/library/ctrl.js [hunk 10]
```diff
@@ -68,18 +75,42 @@
...
-        vm.showSendPopup = false;
+    // Templates
+    vm.deleteTemplate = function (id) {
+      libraryService.deleteTemplate(id).then(function() {
+        getTemplates();
       });
     }
+
+    vm.useTemplate = function (id) {
+      $location.path('template/null/' + id);
```
Comment (dmitrij-borchuk) [inline dev/library/ctrl.js]: What for 'null'?

File: dev/library/ctrl.js [hunk 11]
```diff
@@ -68,18 +75,42 @@
...
+    function getVideos() {
+      libraryService.getVideos().then(function (res) {
+        vm.videosList = res.data.data;
+      });
+    };
+
+    function getTemplates() {
+      libraryService.getTemplates().then(function (res) {
+        vm.templatesList = res.data;
+        console.log(vm.templatesList);
```
Comment (dmitrij-borchuk) [inline dev/library/ctrl.js]: Remove

File: dev/library/ctrl.js [hunk 12]
```diff
@@ -68,18 +75,42 @@
...
       });
     }
+
+    vm.useTemplate = function (id) {
+      $location.path('template/null/' + id);
+    }
+
+    function getVideos() {
+      libraryService.getVideos().then(function (res) {
+        vm.videosList = res.data.data;
```
Comment (dmitrij-borchuk) [inline dev/library/ctrl.js]: data.data ???

File: dev/library/tpl.pug [hunk 13]
```diff
@@ -4,39 +4,31 @@
...
-        type="email"
-        placeholder="email"
-        ng-model="email")
-      .button-container
-        button.close(ng-click="vm.closeSendPopup()") close
-        button.save(ng-click="vm.sendVideo(email, vm.videoForSending)") send
+  template-list(
+    ng-if="vm.showTemplates"
+    list="vm.templatesList"
+    use="vm.useTemplate($data)"
```
Comment (dmitrij-borchuk) [inline dev/library/tpl.pug]: What is 'use'?

File: dev/template/ctrl.js [hunk 14]
```diff
@@ -0,0 +1,189 @@
...
+    var vm = this;
+    vm.name = {};
+    vm.companyRole = {};
+    vm.message = {};
+    vm.title = {};
+    vm.showSendPopup = false;
+    vm.showSelectPopup = false;
+    vm.showSendHandler = false;
+
+    if ($routeParams.templateId === 'null') {
```
Comment (dmitrij-borchuk) [inline dev/template/ctrl.js]: String 'null'???

File: dev/template/ctrl.js [hunk 15]
```diff
@@ -0,0 +1,189 @@
...
+        vm.media = {
+          sources: [{
+            src: res.data.data.attributes.url,
+            type: 'video/mp4'
+          }]
+        };
+      });
+    }
+
+    function checkName() {
```
Comment (dmitrij-borchuk) [inline dev/template/ctrl.js]: Should return the result

**Review by dmitrij-borchuk — CHANGES_REQUESTED** [review]

**Comment by dmitrij-borchuk** [comment]
Please merge dev to this branch

File: dev/library/ctrl.js (hunk 12 above)
Comment (FeditskiyRoman) [inline dev/library/ctrl.js]: response from server is data: {data: {video}}

File: dev/config.js (hunk 6 above)
Comment (RomanFedytskyi) [reply dev/config.js]: I need one url for two differents page, if :id===null i show one page, if templateId===null i show another

File: dev/library/ctrl.js (hunk 12 above)
Comment (RomanFedytskyi) [reply dev/library/ctrl.js]: fixed

Comment (dmitrij-borchuk) [reply dev/library/ctrl.js]: it is still here

**Review by dmitrij-borchuk — CHANGES_REQUESTED** [review]

File: dev/library/ctrl.js [hunk 16]
```diff
@@ -79,7 +76,41 @@
 
     // Send
     vm.sendClickHandler = function (video) {
-      $location.path('template/' + video.id);
+      $location.path('template/' + video.id + '/null');
```
Comment (dmitrij-borchuk) [inline dev/library/ctrl.js]: What for 'null'?

**Review by dmitrij-borchuk — CHANGES_REQUESTED** [review]

**Review by dmitrij-borchuk — APPROVED** [review]

**Pull request merged**